### ISSUE TITLE
Insufficient RAM

### ISSUE THREAD

**Issue by marco74** [jitsi/docker-jitsi-meet#1808]
Hello,

Docker is designed to encapsulate e.g. all those specific options that developers of a specific language think of do not need to be considered by admins. I just started a Jitsi installation using `web-1.0.8000-1`, `prosody-0.12.4`, `jicofo-1.0-1081-1` and `jvb-2.3-109-gacf024b7-1`. The machine has 32 GB RAM and 16 cores. Except Jitsi there's hardly anything running there. There are enough ressources. Although it complains `There is insufficient memory for the Java Runtime Environment to continue.`.

The problem occurs in the containers `jicofo_1` and `jvb_1`, `prosody_1` and `web_1` do not complain. I also tried with `jicofo-1.0-1079-1` and `jvb-2.3-101-g37674374-1` – same complains. Where is the problem? Please fix this problem. Jitsi must also be able to run on such a machine. It did well in corona times > 30 participants in one meeting running on that machine, Why is this not sufficient any more?

Regards

Marco

**Comment by saghul** [comment]
> all those specific options that developers of a specific language think of do not need to be considered by admins

Absolutely not. You can't ignore the requirements of the software just because you run it in a container.

> There are enough ressources. Although it complains `There is insufficient memory for the Java Runtime Environment to continue.`.

How much RAM is allocated to Docker? 

As a starting point I'd suggest you use a release tag, be that stable-XXXX or unstable-XXXX picking random images is not something we support.

**Comment by marco74** [comment]
Sorry I couldn't see that there's also stable, because the list contains a lot of unstable versions. I considered version numbers without `unstable` as `stable`. Now I'm using

* `jitsi/web:stable`
* `jitsi/prosody:stable`
* `jitsi/jicofo:stable`
* `jitsi/jvb:stable`

Concerning limits: here's `docker stats`

![image](https://github.com/jitsi/docker-jitsi-meet/assets/3092437/5f74386a-f494-4077-93dc-ab98e7a7eefe)

The rest of the containers is not even using 16 GB so there's enough RAM left for JVB.

> Absolutely not. You can't ignore the requirements of the software just because you run it in a container.

That's not what I ment. Docker is designed to encapsulate as much as possible to keep as many configuration option away from a system administrator. System Administrators can do a lot of things but they cannot be an expert on each programming language a developer thinks of.

The problem still occurs for jvb and jicofo.

**Comment by saghul** [comment]
This is very odd, I have never seen it.

What Docker version are you using? Did you make any changes to the desafile configuration? What is your base OS?

**Comment by marco74** [comment]
Thank you for your reply.

Now I have the problem. After your reply I tried out various stable versions and found out that `stable-9364-1` worked and `stable-9457-1` not. Based on that and on the fact that hub.docker.com orders the tags/images chronologically, I also tried the unstable version between them. So I found out that `unstable-2024-03-22` worked, but `unstable-2024-03-23` not.

So I looked into the commits. There were three commits right after the stable version 9364-1. One of them looked very interesting: The switch from `bullseye` to `bookworm`. So I tried to build the images right before that commit and this worked. Building the next image failed because GPG verification of `apt-get` used while building the image.

Looking up this I found [that](https://serverfault.com/questions/1137215/how-can-i-write-a-dockerfile-based-on-debian-slim-in-which-apt-get-update-does). Yeah, indeed my server could have an upgrade. But before doing so I tried:

```bash
docker run -ti debian:bookworm-slim /bin/bash -c "apt-get update"
```

what led to similar problems. After upgrading Debian I retried that command and this then worked. Now also `stable-9457` works fine.

Thanks for your help @saghul, and sorry for being rude. Java and me never had been best friends mostly because Java always uses to much RAM, now that integrated into many other things. Sorry for that, and thank you for your help.

Marco
